### DISCUSSION TITLE
Regarding the PEM data decoding

### DISCUSSION THREAD

**Discussion by ranjithravitk** [openssl-projects/pkcs11-provider#414] [Q&A]
Hi, I am currently implementing the elgamal provider for the openssl and I am facing an issue while doing so. I have successfully encoded the Privatekey in PEM format and have successfully implemented the decoder as well. But when I pass back the key object reference to the OpenSSL and since it is using the default "STORE" which is used for file scheme, it is not creating the pkey as the file "STORE" doesn't support export function. I just wanted to ask how you guys handled the pem data reading and generated the public key from it. Any tips will be really helpful. Thanks you

**Comment by simo5** [comment]
We allow to export public keys but not private keys, because generally tokens simply can't export them. Although we do have a faux encoder for the "Private key as a URI" option.

So I gues you'll have to be more precise in what you are looking for, what token do you have that support this ancient broken encryption scheme?
